api/v1alpha1/nodepool_types.go

```diff
@@ -263,7 +263,6 @@ type NodePoolManagement struct {
 	// Replace is the configuration for rolling upgrades.
 	//
 	// +kubebuilder:validation:Optional
-	// +kubebuilder:default={strategy: "RollingUpdate", rollingUpdate: {maxSurge: 1, maxUnavailable: 0 }}
```

Comment:
what will the capi machinedeployment controller do if these are unset?

Reply:
Okay, then the `optional` is wrong now?

Reply:
no because it can be empty when UpgradeType=InPlace

Reply:
@alvaroaleman Let's proceed with this PR and address the webhook separately following up to https://github.com/openshift/hypershift/pull/1346 so the validation logic can be reused between backend and webhook.

Reply:
the maxSurge and maxUnavailable fields themselves still document default values

https://github.com/openshift/hypershift/blob/eaea63a0759260318fbb0ea35d24f9243223aa7f/api/v1alpha1/nodepool_types.go#L218

https://github.com/openshift/hypershift/blob/eaea63a0759260318fbb0ea35d24f9243223aa7f/api/v1alpha1/nodepool_types.go#L239

were these default values expected to come from this kubebuilder tag that's being removed?

Reply:
Thanks David, added the defaults for those.

Reply:
So now what happens if someone sets the upgrade type to `replace` but doesn't explicilty set the `replace` field? Will it get defaulted somehow. What would be the expectation?

Reply:
It must fail since those are required fields, we'll add that defaulting to the optional webhook though.